src/resolve/resolved-dns-rr.c

```diff
@@ -2097,23 +2097,27 @@ DnsSvcParam *dns_svc_params_copy(DnsSvcParam *first) {
         return copy;
 }
 
-int dns_txt_item_new_empty(DnsTxtItem **ret) {
+
+int dns_txt_item_new(DnsTxtItem **ret, size_t len) {
         DnsTxtItem *i;
 
         assert(ret);
 
-        /* RFC 6763, section 6.1 suggests to treat
-         * empty TXT RRs as equivalent to a TXT record
-         * with a single empty string. */
-
-        i = malloc0(offsetof(DnsTxtItem, data) + 1); /* for safety reasons we add an extra NUL byte */
+        i = malloc0(offsetof(DnsTxtItem, data) + len + 1); /* for safety reasons we add an extra NUL byte */
```

Comment:
maybe, if you have an explicit constructor for this, maybe also add an overflow check?

i.e. 

```c
if (len > SIZE_MAX - offsetof(DnsTxtItem, data) - 1)
        return -ENOMEM;
```

Reply:
BTW.. the actual limit is PTRDIFF_MAX not SIZE_MAX. neither clang nor GCC know how to do pointer math with objects bigger than that and the code generated if any is undefined. (and it wont be fixed)

Reply:
we are not doing pointer math here though? just integers?